doc/components/popup.md

```diff
@@ -0,0 +1,481 @@
+size = 50,
+```
+
+```lua
+size = "50%",
+```
+
+```lua
+size = {
+    width = 30,
+    height = 50,
+}
+```
+
+```lua
+size = {
+    width = "30%",
+    height = "50%",
+}
+```
+
+---
+
+### `enter`
+
+- **Type:** `boolean`
+- **Description:** If `true`, the popup is auto-entered after mount
+
+#### Usage
+
+```lua
+enter = true,
+```
+
+---
+
+### `focusable`
+
+- **Type:** `boolean`
+- **Description:** Is `false`, the popup can not be entered by user actions (wincmds, mouse events)
+
+#### Usage
+
+```lua
+focusable = true,
+```
+
+---
+
+### `zindex`
+
+- **Type:** `number`
+- **Description:**
+  - Sets the order of the popup on z-axis
+  - Popup with higher the `zindex` goes on top of popups with lower `zindex`
+
+#### Usage
+
+```lua
+zindex = 20,
+```
+
+---
+
+### `buf_options`
+
+- **Type:** `table`
+- **Description:** Contains all buffer related options
+
+#### Usage
+
+```lua
+buf_options = {
+    modifiable = false,
+    readonly = true,
+},
+```
+
+#### Descriptions
+
+| Value        | Description                                                                  |
+| ------------ | ---------------------------------------------------------------------------- |
+| `modifiable` | If `false` the buffer contents cannot be changed. <br> Check `:h modifiable` |
+| `readonly`   | If `true`, writes fail unless you use a `!` <br> Check `:h readonly`         |
+
+---
+
+### `win_options`
+
+- **Type:** `table`
+- **Description:** Contains all window related options
+
+#### Usage
+
+```lua
+win_options = {
+    winblend = 10,
+    winhighlight = "Normal:Normal",
+},
+```
+
+#### Descriptions
+
+| Value          | Description                                                                               |
+| -------------- | ----------------------------------------------------------------------------------------- |
+| `winblend`     | Transparency for a floating window. <br> Goes from `0` to `100`. <br> Check `:h winblend` |
+| `winhighlight` | Window local highlights. <br> Check `:h winhighlight`                                     |
+
+## Method Descriptions
+
+### `popup:mount()`
+
+- Mounts the popup
+
+```lua
+popup:mount()
+```
+
+---
+
+### `popup:unmount()`
+
+- Unmounts the popup
+
+```lua
+popup:unmount()
+```
+
+---
+
+### `popup:map(mode, key, handler, opts, force)`
+
+- Sets keymap for this popup.
+- If keymap was already set and `force` is not `true` it returns `false`, otherwise returns `true`
+
+```lua
+local ok = popup:map("n", "<esc>", function(bufnr)
+    print("ESC pressed in Normal mode!")
+end, { noremap = true })
+```
+
+#### Descriptions
+
+| Value     | Description                           |
+| --------- | ------------------------------------- |
+| `mode`    | Neovim `mode` <br> See `:h vim-modes` |
+| `key`     | Trigger for the map                   |
+| `handler` | Function to fire on that key trigger  |
+| `opts`    | TODO: what goes here                  |
+| `force`   | TODO: what goes here                  |
```

Comment:
`force` is a boolean: `true`/`false`

https://github.com/MunifTanjim/nui.nvim/blob/abd1a4a23d0bbc0d0d4bc8faaa240330b44a4d5c/lua/nui/popup/init.lua#L266